pkg/plugin/aggregation/aggregator_test.go

```diff
@@ -286,56 +289,58 @@ func TestAggregation_RetryWindow(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		agg := NewAggregator(dir, expected)
-		// Shorten retry window for testing.
-		agg.retryWindow = testRetryWindow
-		testDataPath := "./testdata/fakeLogData.txt"
-		testDataReader, err := os.Open(testDataPath)
-		if err != nil {
-			t.Fatalf("Could not open test data file: %v", err)
-		}
-		defer testDataReader.Close()
-
-		var r io.Reader
-		if tc.simulateErr {
-			r = iotest.TimeoutReader(testDataReader)
-		} else {
-			r = strings.NewReader("foo")
-		}
+		t.Run(tc.desc, func(t *testing.T) {
+			agg := NewAggregator(dir, expected)
+			// Shorten retry window for testing.
+			agg.retryWindow = testRetryWindow
+			testDataPath := "./testdata/fakeLogData.txt"
+			testDataReader, err := os.Open(testDataPath)
+			if err != nil {
+				t.Fatalf("Could not open test data file: %v", err)
+			}
+			defer testDataReader.Close()
 
-		err = agg.processResult(&plugin.Result{Body: r, NodeName: "node1", ResultType: "systemd_logs"})
-		if err == nil && tc.simulateErr {
-			t.Fatal("Expected error processing this due to reading error, instead got nil.")
-		}
-		// check time before/after wait and ensure it is greater than the retryWindow.
-		time.Sleep(tc.postProcessSleep)
-		start := time.Now()
-		agg.Wait(make(chan bool))
-		waitTime := time.Now().Sub(start)
+			var r io.Reader
+			if tc.simulateErr {
+				r = iotest.TimeoutReader(testDataReader)
+			} else {
+				r = strings.NewReader("foo")
+			}
 
-		// Add buffer to avoid raciness due to processing time.
-		diffTime := waitTime - tc.expectExtraWait
-		if diffTime > testBufferDuration || diffTime < -1*testBufferDuration {
-			t.Errorf("Expected Wait() to wait the duration (%v) due to failed result, instead waited only %v", agg.retryWindow, waitTime)
-		}
+			err = agg.processResult(&plugin.Result{Body: r, NodeName: "node1", ResultType: "systemd_logs"})
+			if err == nil && tc.simulateErr {
+				t.Fatal("Expected error processing this due to reading error, instead got nil.")
+			}
+			// check time before/after wait and ensure it is greater than the retryWindow.
+			time.Sleep(tc.postProcessSleep)
+			start := time.Now()
+			agg.Wait(make(chan bool))
+			waitTime := time.Now().Sub(start)
+
+			// Add buffer to avoid raciness due to processing time.
+			diffTime := waitTime - tc.expectExtraWait
+			if diffTime > testBufferDuration || diffTime < -1*testBufferDuration {
+				t.Errorf("Expected Wait() to wait the duration %v (+/- %v), instead waited %v", tc.expectExtraWait, testBufferDuration, waitTime)
```

Comment:
Cleaned up the error message; it was confusing in cases 
 - where we weren't causing errors
 - where the wait time > expectedWaitTime
 - what precision we expected